src/backend/distributed/executor/multi_utility.c

```diff
@@ -904,6 +933,61 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 
 
 /*
+ * ErrorIfDistributedAlterSeqOwnedBy errors out if the provided alter sequence
+ * statement attempts to change the owned by property of a distributed sequence
+ * or attempt to change a local sequence to be owned by a distributed table.
+ */
+static void
+ErrorIfDistributedAlterSeqOwnedBy(AlterSeqStmt *alterSeqStmt)
+{
+	Oid sequenceId = RangeVarGetRelid(alterSeqStmt->sequence, AccessShareLock,
+	                                  alterSeqStmt->missing_ok);
+	Oid ownedByTableId = InvalidOid;
+	int32 ownedByColumnId = 0;
+	bool hasDistributedOwner = false;
+	ListCell *optionCell = NULL;
+
+	/* alter statement referenced nonexistent sequence; return */
+	if (sequenceId == InvalidOid)
+	{
+		return;
+	}
+
+	if (sequenceIsOwned(sequenceId, &ownedByTableId, &ownedByColumnId))
+	{
+		hasDistributedOwner = IsDistributedTable(ownedByTableId);
+	}
+
+	foreach(optionCell, alterSeqStmt->options)
+	{
+		DefElem *defElem = (DefElem *) lfirst(optionCell);
+		if (strcmp(defElem->defname, "owned_by") == 0)
+		{
+			List *ownedByNames = defGetQualifiedName(defElem);
+			int nameCount = list_length(ownedByNames);
+
+			if (hasDistributedOwner)
+			{
```

Comment:
We should only error out if the owner is different, lots of tools just specify all the options.